### PR TITLE
Add unicode-hashtag module for hashtag regex

### PR DESCRIPTION
**Pull request by christianbundy**
Hey @qubist! This adds the latest commits from ssbc/ssb-markdown master as well as a commit to use the hashtag-regex module for hashtag parsing. Please let me know if you'd like to see any changes!

### CODE REVIEW

**Comment by christianbundy** [comment]
(The relevant commit is https://github.com/qubist/ssb-markdown/pull/1/commits/232c9443c90c0bb8434b356495dba8de8e40657a)

File: lib/block.js [hunk 1]
```diff
@@ -4,6 +4,19 @@ const hljs = require('highlight.js')
 const util = require('./util')
 const emojiByShortcode = require('markdown-it-emoji/lib/data/full.json')
 const emojiByCharacter = util.reverseKeyValue(emojiByShortcode)
+const hashtagRegex = require('hashtag-regex')
+
+// The hashtag-regex module comes with a regex that auto-detects the number sign
+// characters, but markdown-it-hashtag wants to handle that by itself. This
+// function strips the # detection from the hashtag-regex output and so that it
```
Comment (qubist) [inline lib/block.js]:
```suggestion
// function strips the # detection from the hashtag-regex output so that it
```

**Review by qubist — CHANGES_REQUESTED** [review]
Everything looks great besides this lil typo!

**Comment by qubist** [comment]
Ok I tested this by setting up my development version of Patchwork (up-to-date) to use this code for ssb-markdown by changing this line in package.json and re-npm installing/running:
```
"ssb-markdown": "github:ssbc/ssb-markdown#unicode-hashtag",
```
and I'm seeing this result when I preview a post:
Screen Shot 2019-08-19 at 3 28 36 PM

This looks great in that `#轻拿轻放`, `#malmo`, and `#poŝtelefono` are linked, but I see two potential problems: `#dash-between` and `#plus+between`. In my reading of the Unicode 	Hashtag Identifier Syntax, these should count as hashtags and be linked:

> Continue := XID_Continue, plus Emoji, Emoji_Component, and “_”, “-”, “+”, minus Start characters. 

([source](http://unicode.org/reports/tr31/#hashtag_identifiers))
@christianbundy Thoughts? It's possible I'm making a mistake in how I'm testing this or how I'm interpreting the Unicode spec.

File: lib/block.js (hunk 1 above)
Comment (christianbundy) [reply lib/block.js]: Nice, thanks!

**Comment by qubist** [comment]
Upon further investigation, this looks like an issue (or not an issue?) with the hashtag-regex package. I cloned that repo and added tests for 'a-b-c' and 'a+b+c' (it already had and was passing 'a_b_c').

Results of `npm test`:
```
  1) Hashtag identifier regex
       matches #a+b+c as a single unit:

      AssertionError [ERR_ASSERTION]: Expected values to be loosely deep-equal:

'＃a'

should loosely deep-equal

'＃a+b+c'
      + expected - actual

      -＃a
      +＃a+b+c



  3) Hashtag identifier regex
       matches #a-b-c as a single unit:

      AssertionError [ERR_ASSERTION]: Expected values to be loosely deep-equal:

'＃a'

should loosely deep-equal

'＃a-b-c'
      + expected - actual

      -＃a
      +＃a-b-c
```

**Comment by christianbundy** [comment]
Here's what I'm seeing:

Screenshot from 2019-08-19 12-47-59

> In my reading of the Unicode Hashtag Identifier Syntax, these should count as hashtags and be linked:

I feel the same way. I'm not sure why it's doing that, I'll investigate a bit.

**Review by qubist — APPROVED** [review]

**Pull request merged**